plugins/automation/lib/discourse_automation/triggers/recurring.rb

```diff
@@ -87,12 +87,12 @@ def self.setup_pending_automation(automation, fields, previous_fields)
             RRule::Rule
               .new("FREQ=MONTHLY;INTERVAL=#{interval};BYDAY=#{count}#{byday}", dtstart: start_date)
               .between(Time.now, interval_end.months.from_now)
-              .first
+              .find { |date| date > Time.zone.now }
           when "year"
             RRule::Rule
               .new("FREQ=YEARLY;INTERVAL=#{interval}", dtstart: start_date)
               .between(Time.now, interval_end.years.from_now)
-              .first
+              .find { |date| date > Time.zone.now }
           end
 
         if next_trigger_date && next_trigger_date > Time.zone.now
```

Comment:
We have this `if` condition to ensure that `next_trigger_date` is actually ahead of the current time. If it's not, then we give up, therefore the automation ends up with no pending automation records, which means it has stalled.